staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -91,8 +60,8 @@ var _ http.Hijacker = &fancyResponseWriterDelegator{}
 // 2. the response line containing:
 //    - the unique id from 1
 //    - response code
-func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, out io.Writer) http.Handler {
-	if out == nil {
+func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, sink audit.Sink, policy *auditinternal.Policy, longRunningCheck filters.LongRunningRequestCheck) http.Handler {
```

Comment:
done, moved all non-plumbing code to pkg/audit, but left the filter with the WriteHeader intercept logic in filters. I think that's a sensible split.